spec/workers/event_worker_spec.rb

```diff
@@ -14,7 +17,47 @@
   #   end.to change(described_class.jobs, :size).by(1)
   # end
 
-  pending 'can process an event'
-  pending 'can retry 3 times'
-  pending 'can clean up when errors are raised'
+  before do
+    stub_request(:post, 'https://myfakeoutbound.com/')
+      .with(headers: { 'Accept' => '*/*', 'User-Agent' => 'Ruby' })
+      .to_return(status: 200, body: { data: 'stubbed response' }.to_json, headers: {})
+  end
+
+  context 'Event worker' do
+    let(:event) { create :event }
+
+    it 'can process an event' do
+      expect(event.reload.completed).to eq false
+
+      EventWorker.new.perform event.id
+
+      expect(event.reload.completed).to eq true
+    end
+
+    it 'can retry 3 times' do
+      worker = EventWorker.new
+      req_handler = ::BridgeApi::Http::RequestHandler.new event
+      req_handler.formatter = MockFailFormatter.new
+      worker.request_handler = req_handler
+
+      expect do
+        worker.perform event.id
+      end.to raise_error StandardError
+
+      expect(EventWorker.jobs.count).to eq 1
+      # TODO: - Need to fix retry_count bug
+    end
+
+    it 'can clean up when errors are raised' do
+      worker = EventWorker.new
+      req_handler = ::BridgeApi::Http::RequestHandler.new event
+      req_handler.formatter = MockFailFormatter.new
+      worker.request_handler = req_handler
+
+      expect do
+        worker.perform event.id
+      end.to raise_error StandardError
+      expect(JSON.parse(event.reload.data)['outbound'][0]['response']).to eq({ 'message' => 'StandardError' })
```

Comment:
You should add an expect for request as well. Look at my formatter spec. You will need to add like 3-4 expects for individual values because you can't do the whole object due to timestamps.